app/src/main/java/com/example/admin/personallibrarycatalogue/BooksListActivityFragment.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Objects;
+
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+public class BooksListActivityFragment extends Fragment {
+
+    private final static String ID = "id";
+    private ListView listView_;
+    private BooksListAdapter booksListAdapter_;
+    private LibraryDatabaseHelper helper_;
+
+    public BooksListActivityFragment() {
+    }
+
+    @Override
+    public View onCreateView(final LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_books_list, container, false);
+
+        helper_ = new LibraryDatabaseHelper(getActivity());
+
+        List<Book> booksList = new ArrayList<Book>();
+        booksList = helper_.getAllBooks();
+
+        listView_ = (ListView) rootView.findViewById(R.id.books_list_view);
+        booksListAdapter_ = new BooksListAdapter(this.getActivity(), booksList);
+        listView_.setAdapter(booksListAdapter_);
+
+        // Set ContextMenu for listView
+        registerForContextMenu(listView_);
+
+        listView_.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                changeBook(parent, position);
+            }
+        });
+
+        return rootView;
+    }
+
+    @Override
+    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
+        super.onCreateContextMenu(menu, v, menuInfo);
+        MenuInflater inflater = getActivity().getMenuInflater();
+        menu.setHeaderTitle(R.string.Menu);
+        inflater.inflate(R.menu.context_menu, menu);
+    }
+
+    @Override
+    public boolean onContextItemSelected(MenuItem item) {
+        AdapterView.AdapterContextMenuInfo info = (AdapterView.AdapterContextMenuInfo) item.getMenuInfo();
+
+        switch (item.getItemId()) {
+            case R.id.edit_book:
+                changeBook(listView_, info.position);
+                break;
+
+            case R.id.delete_book:
+                deleteBook(listView_, info.position);
+                booksListAdapter_.notifyDataSetChanged();
+                break;
+
+            default:
+                return super.onContextItemSelected(item);
+        }
+        return true;
+    }
+
+    /**
+     * Set up activity for changing some information about book (title, author etc.)
+     *
+     * @param parent   AdapterView (could be a ListView, GridView, Spinner)
+     * @param position Clicked item position
+     */
+    public void changeBook(AdapterView<?> parent, int position) {
+        Book book = (Book) parent.getItemAtPosition(position);
+        String author = book.getAuthor();
+        String title = book.getTitle();
+
+        Intent intent = new Intent();
+        intent.setClass(getActivity(), AddBookActivity.class);
+
+        int returned_id = helper_.getId(title, author);
```

Comment:
UPDATE: you have id field in Book. Why not to use it instead of fetched one? Also your way of getting id contains a problem: if you have two books with the same title and author which id it will return?